api/src/main/java/com/codeforcommunity/rest/subrouter/TeamsRouter.java

```diff
@@ -22,12 +31,143 @@ public TeamsRouter(ITeamsProcessor processor) {
+    JWTData userData = ctx.get("jwt_data");
+    CreateTeamRequest createTeamRequest =
+        RestFunctions.getJsonBodyAsClass(ctx, CreateTeamRequest.class);
+    processor.createTeam(userData, createTeamRequest);
+    end(ctx.response(), 200);
+  }
+
+  // Get a Team
+  private void registerGetTeam(Router router) {
+    Route getTeamRoute = router.get("/:team_id");
+    getTeamRoute.handler(this::handleGetTeamRoute);
+  }
+
+  private void handleGetTeamRoute(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(ctx.request(), "team_id");
+    TeamDataResponse getTeamResponse = processor.getTeam(userData, teamId);
+    end(ctx.response(), 200, JsonObject.mapFrom(getTeamResponse).toString());
+  }
+
+  // Add a Goal
+  private void registerAddGoal(Router router) {
+    Route addGoalRoute = router.post("/:team_id/add_goal");
+    addGoalRoute.handler(this::handleAddGoal);
+  }
+
+  private void handleAddGoal(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(ctx.request(), "team_id");
+    AddGoalRequest addGoalRequest =
+        RestFunctions.getJsonBodyAsClass(ctx, AddGoalRequest.class);
+    processor.addGoal(userData, addGoalRequest, teamId);
+    end(ctx.response(), 200);
+  }
+
+  // Delete a Goal
+  private void registerDeleteGoal(Router router) {
+    Route deleteGoalRoute = router.post("/:team_id/delete_goal/:goal_id");
+    deleteGoalRoute.handler(this::handleDeleteGoal);
+  }
+
+  private void handleDeleteGoal(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(ctx.request(), "team_id");
+    int goalId = RestFunctions.getRequestParameterAsInt(ctx.request(), "goal_id");
+    processor.deleteGoal(userData, teamId, goalId);
+    end(ctx.response(), 200);
+  }
+
+  private void registerInviteUser(Router router) {
+    Route inviteUserRoute = router.post("/:team_id/invite");
+    inviteUserRoute.handler(this::handleInviteUser);
+  }
+
+  private void handleInviteUser(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(ctx.request(), "team_id");
+    InviteUsersRequest inviteUserRequest =
+        RestFunctions.getJsonBodyAsClass(ctx, InviteUsersRequest.class);
+    processor.inviteUser(userData, inviteUserRequest, teamId);
+    end(ctx.response(), 200);
+  }
+
+  private void registerGetApplicants(Router router) {
+    Route getApplicantsRoute = router.get("/:team_id/applicants");
+    getApplicantsRoute.handler(this::handleGetApplicantsRoute);
+  }
+
+  private void handleGetApplicantsRoute(RoutingContext ctx) {
+    JWTData userData = ctx.get("jwt_data");
+    int teamId = RestFunctions.getRequestParameterAsInt(ctx.request(), "team_id");
+    List<UsersTeamDataResponse> response = processor.getApplicants(userData, teamId);
```

Comment:
The API response for this route is
```
{
  "applicants": [
    {
      "userId": INT,
      "username": STRING,
    },
    ...
  ]
}
```
Based on that, you need to be returning a single object that has a single field called applicants that is a list of UsersTeamDataResponse objects. With the setup here, you're just returning a JSON Array. Of course currently UsersTeamDataResponse has the wrong fields but we already said we needed to add a TODO for that.

Reply:
TODO and monday ticket for this